src/readabilityScore/IndexCalculator.java

```diff
@@ -0,0 +1,20 @@
+package readabilityScore;
+
+ /*
+ Methods should be used in the order:
+
+ setText(Text text),
+ calculateIndex(),
+ determineAge(),
+ displayInformation()
+
+ to initialize all fields and receive correct result
+  */
+
+interface IndexCalculator {
+    void setText(Text text);
+    void calculateIndex();
+    Integer determineAge();
+    void displayInformation();
```

Comment:
This interface is not bad, but can be simplified further?

Now the user of this interface should make sure that the methods are invoked in particular order. Let's think about if we could design a better interface where we wouldn't need to remember in which order we should invoke the methods.

It looks like we could have two methods:

```
interface IndexCalculator {
    double calculateIndex(Text text);
    Index index();
}
```

This interface is easier to use, we don't need to write a comment about how it should be used.
Usually, if we have to write a comment, it indicates that something can be improved: https://critter.blog/2020/09/15/dont-comment-your-code-refactor-it/